apps/microtvm/arduino/template_project/microtvm_api_server.py

```diff
@@ -248,7 +256,12 @@ def _convert_includes(self, project_dir, source_dir):
         for ext in ("c", "h", "cpp"):
             for filename in source_dir.rglob(f"*.{ext}"):
                 with filename.open() as file:
-                    lines = file.readlines()
+                    try:
+                        lines = file.readlines()
+                    # TODO: This exception only happens using `tvmc micro` and is not catched on Arduino tests.
+                    # Needs more investigation.
+                    except UnicodeDecodeError:
```

Comment:
do we know more about this or have a stacktrace? I think best fix if we can't work around the error would be to open the file in binary mode (`filename.open("rb")`) and then below:
```
for i, line in in enumerate(lines):
  try:
    line_str = str(line, "utf-8")
  except UnicodeDecodeError:
    continue
  result = re.search(r"#include...", line_str)
  # ...
```

Reply:
done